consensus/core/src/transaction.rs

```diff
@@ -37,6 +38,18 @@ pub(crate) struct TransactionConsumer {
     max_consumed_bytes_per_request: u64,
     max_consumed_transactions_per_request: u64,
     pending_transactions: Option<TransactionsGuard>,
+    block_status_subscribers: Arc<Mutex<BTreeMap<BlockRef, Vec<oneshot::Sender<BlockStatus>>>>>,
```

Comment:
I did consider using `NotifyRead` instead - I've even went ahead in the beginning and implemented with it . But 
1. `NotifyRead` allow us to notify waiters in a strictly `key` fashion. Here for gc'ed blocks I would just like to notify everyone who is bellow the `gc_round` (see the implementation below) . I could still do it by keeping a secondary index (ex a `BTreeSet` ) with all the submitted blocks, but then I started seeing less value on using `NotifyRead`
2. The data are quite ephemeral and this map should get cleaned up quickly. The only case where I see something pathogenic happening is for nodes which somehow (a) produce bunch of blocks , (b) never commit/gc & (c) never see their blocks getting committed. I am not even sure if that's possible with the current code, and a quite edge case to be concerned about